bgpd/bgp_zebra.c

```diff
@@ -1166,6 +1177,10 @@ static bool update_ipv6nh_for_route_install(int nh_othervrf, struct bgp *nh_bgp,
 	if (nexthop)
 		api_nh->gate.ipv6 = *nexthop;
 
+	/* api_nh structure has union of gate and bh_type */
+	if (api_nh->type != NEXTHOP_TYPE_BLACKHOLE)
+		api_nh->gate.ipv6 = *nexthop;
```

Comment:
why do this a second time - the assignment looks just the same just above this change?